lib/matplotlib/backend_bases.py

```diff
@@ -2555,6 +2605,123 @@ class NonGuiException(Exception):
     pass
 
 
+class WindowEvent(object):
+    def __init__(self, name, window):
+        self.name = name
+        self.window = window
+
+
+class WindowBase(cbook.EventEmitter):
```

Comment:
Ah fair enough, if you are combining this with MI with gui stuff then ABC is a non-starter.

I don't think we use ABC else where because it didn't exist when the code was written, but I think is the right solution to use when writing classes that _must_ be subclassed which methods/attributes which _must_ be overridden as it pushes discovering the error is your code up to object instantiation time not object use time.